bot/worker.py

```diff
@@ -0,0 +1,184 @@
+
+    However, just as a safeguard, Caching is used to store thing ids as they are processed for the first time.
+    Otherwise, when the bot is restarted it might reply twice to same comments. If thing id is in the already present
+    in the cache_api, then it is ignored, else processed and added to the cache_api.
+    * Self comments are ignored.
+    * It is prepared for comparision to the responses in dictionary.
+    * If the thing is not on the excluded responses list (loaded from config) and if it is in the responses db or
+    specific responses list, a reply thing is prepared and posted.
+    """
+
+    if cache_api.check(thing_id=thing.fullname):
+        return
+
+    # Ignore thyself
+    if thing.author == reddit.user.me:
+        return
+
+    logger.debug("Found new thing: " + str(thing.fullname))
+
+    processed_body = process_body(thing.body if isinstance(thing, Comment) else thing.title)
+
+    # Don't reply to single word text (they're mostly common phrases).
+    if ' ' not in processed_body:
+        return
+
+    if processed_body in config.EXCLUDED_RESPONSES:
+        return
+
+    if processed_body in config.CUSTOM_RESPONSES:
+        do_custom_reply(thing=thing, custom_response=config.CUSTOM_RESPONSES[processed_body])
+
+    if do_flair_specific_reply(thing, processed_body):
+        return
+
+    do_regular_reply(thing, processed_body)
+
+
+def process_body(body_text):
+    """Method used to clean the thing text. Logic is similar to clean_response_text on wiki parsers.
+    * If thing contains a quote, the first quote is considered as the response_text.
+    * Punctuation marks are replaced with space.
+    * The response_text is turned to lowercase.
+    * Converts multiple spaces into single space.
+
+    Removed code to remove repeating letters in a thing because it does more harm than good - words like 'all',
+    'tree' are stripped to 'al' and 'tre' which dont match with any responses.
+
+    :param body_text: The thing body
+    :return: Processed thing body
+    """
+
+    if '>' in body_text:
+        lines = body_text.split('\n\n')
+        for line in lines:
+            if line.startswith('>'):
+                body_text = line
+                break
+
+    body_text = body_text.translate(PUNCTUATION_TRANS)
+    body_text = body_text.translate(WHITESPACE_TRANS)
+
+    while '  ' in body_text:
+        body_text = body_text.replace('  ', ' ')
+
+    body_text = body_text.strip().lower()
+
+    return body_text
+
+
+def do_flair_specific_reply(thing, response):
+    hero_id = db_api.get_hero_id_by_flair_css(flair_css=thing.author_flair_css_class)
+    if hero_id:
+        link, hero_id = db_api.get_link_for_response(
+            response=response, hero_id=hero_id)
+        if link:
+            reply = create_reply(thing=thing, response_url=link, hero_id=hero_id)
+            thing.reply(reply)
+            logger.info("Added: " + thing.fullname)
+            return True
```

Comment:
I think that method name should imply that a boolean is returned, maybe something like 'flair_specific_reply_added' or 'adding_flair_specific_reply'? Right now I would expect both "do_flair_specific_reply" and "do_regular_reply" behaviours to be the same, based on their names, yet this is not the case.

Reply:
Changed the name to `try_flair_specific_reply` in 41e1416fd2c7051547d1ef4807ac17741893a239